sequencer/src/chain_config.rs

```diff
@@ -0,0 +1,90 @@
+use crate::state::FeeAmount;
+use commit::{Commitment, Committable};
+use derive_more::{From, Into};
+use ethers::types::U256;
+use serde::{Deserialize, Serialize};
+
+#[derive(Default, Hash, Copy, Clone, Debug, Deserialize, Serialize, PartialEq, Eq, From, Into)]
+pub struct ChainId(U256);
+
+impl From<u16> for ChainId {
+    fn from(id: u16) -> Self {
+        Self(id.into())
+    }
+}
+
+impl ChainId {
+    // TODO: this duplicate code, consider a macro for U256 newtypes
+    pub(crate) fn to_fixed_bytes(self) -> [u8; 32] {
+        let mut bytes = [0u8; core::mem::size_of::<U256>()];
+        self.0.to_little_endian(&mut bytes);
+        bytes
+    }
```

Comment:
self.0.to_fixed_bytes()

Reply:
I do think we can add a macro in `commit` to avoid writing these code every time and being non-consistent

Reply:
I added a macro in the sequencer crate: https://github.com/EspressoSystems/espresso-sequencer/pull/1318/commits/037958322a8ee46b444ffd01ce8cb6b5cc8f238a, I'm not sure what exactly you would like to have a macro for in the commit crate.